core/templates/domain/topic/topic-rights-backend-api.service.ts

```diff
@@ -16,162 +16,188 @@
-  '$http', '$q', 'UrlInterpolationService',
-  'TOPIC_RIGHTS_URL_TEMPLATE',
-  function($http, $q, UrlInterpolationService,
-      TOPIC_RIGHTS_URL_TEMPLATE) {
-    // Maps previously loaded topic rights to their IDs.
-    var topicRightsCache = {};
-
-    var _fetchTopicRights = function(topicId, successCallback,
-        errorCallback) {
-      var topicRightsUrl = UrlInterpolationService.interpolateUrl(
-        TOPIC_RIGHTS_URL_TEMPLATE, {
-          topic_id: topicId
-        });
-
-      $http.get(topicRightsUrl).then(function(response) {
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { HttpClient } from '@angular/common/http';
+
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+import { TopicDomainConstants } from
+  'domain/topic/topic-domain.constants';
+import cloneDeep from 'lodash/cloneDeep';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TopicRightsBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolation: UrlInterpolationService) {}
+  // Maps previously loaded topic rights to their IDs.
+  private topicRightsCache = {};
+
+  private _fetchTopicRights(
+      topicId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    var topicRightsUrl = this.urlInterpolation.interpolateUrl(
+      TopicDomainConstants.TOPIC_RIGHTS_URL_TEMPLATE, {
+        topic_id: topicId
+      });
+
+    this.http.get(
+      topicRightsUrl, { observe: 'response' }).toPromise().then(
+      (response) => {
         if (successCallback) {
-          successCallback(response.data);
+          successCallback(response.body);
         }
-      }, function(errorResponse) {
+      }, (error) => {
         if (errorCallback) {
-          errorCallback(errorResponse.data);
+          errorCallback(error.data);
         }
       });
-    };
+  }
 
-    var _setTopicStatus = function(
-        topicId, publishStatus, successCallback, errorCallback) {
-      var changeTopicStatusUrl = UrlInterpolationService.interpolateUrl(
-        '/rightshandler/change_topic_status/<topic_id>', {
-          topic_id: topicId
-        });
+  private _setTopicStatus(
+      topicId: string,
+      publishStatus: boolean,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    var changeTopicStatusUrl = this.urlInterpolation.interpolateUrl(
+      '/rightshandler/change_topic_status/<topic_id>', {
+        topic_id: topicId
+      });
 
-      var putParams = {
-        publish_status: publishStatus
-      };
+    var putParams = {
+      publish_status: publishStatus
+    };
 
-      $http.put(changeTopicStatusUrl, putParams).then(function(response) {
-        topicRightsCache[topicId] = response.data;
+    this.http.put(
+      // eslint-disable-next-line max-len
```

Comment:
This is not needed (You can split the next line)

Reply:
Done